Cubical/HITs/Rationals/QuoQ/Properties.agda

```diff
@@ -0,0 +1,245 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.HITs.Rationals.QuoQ.Properties where
+
+open import Cubical.Foundations.Everything hiding (_⁻¹)
+
+open import Cubical.HITs.Ints.QuoInt as ℤ using (ℤ; Sign; signed; pos; neg; posneg; sign)
+open import Cubical.HITs.SetQuotients as SetQuotient using () renaming (_/_ to _//_)
+
+open import Cubical.Data.Nat as ℕ using (ℕ; zero; suc)
+open import Cubical.Data.NatPlusOne
+open import Cubical.Data.Sigma
+
+open import Cubical.Data.Sum
+open import Cubical.Relation.Nullary
+
+open import Cubical.HITs.Rationals.QuoQ.Base
+
+ℚ-cancelˡ : ∀ {a b} (c : ℕ₊₁) → [ ℕ₊₁→ℤ c ℤ.* a / c *₊₁ b ] ≡ [ a / b ]
+ℚ-cancelˡ {a} {b} c = eq/ _ _
+  (cong (ℤ._* ℕ₊₁→ℤ b) (ℤ.*-comm (ℕ₊₁→ℤ c) a) ∙ sym (ℤ.*-assoc a (ℕ₊₁→ℤ c) (ℕ₊₁→ℤ b)))
+
+ℚ-cancelʳ : ∀ {a b} (c : ℕ₊₁) → [ a ℤ.* ℕ₊₁→ℤ c / b *₊₁ c ] ≡ [ a / b ]
+ℚ-cancelʳ {a} {b} c = eq/ _ _
+  (sym (ℤ.*-assoc a (ℕ₊₁→ℤ c) (ℕ₊₁→ℤ b)) ∙ cong (a ℤ.*_) (ℤ.*-comm (ℕ₊₁→ℤ c) (ℕ₊₁→ℤ b)))
+
+-- useful functions for defining operations on ℚ
+
+onCommonDenom :
```

Comment:
Wow, this proof is so scary... Makes me think of Brainfuck programming :-)

Reply:
Lots of these proofs are very scary looking but have very little content... I wonder if #297 could be used here .